authentication/views.py

```diff
@@ -1,13 +1,58 @@
-from django.shortcuts import render
+from django.shortcuts import render,redirect    
 from django.contrib.auth import login,logout,authenticate
+from django.http import HttpResponseRedirect, HttpResponse
+from django.urls import reverse
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.models import User
 # Create your views here.
 
 
 def loginView(request):
-    pass
-
+    if request.method == 'POST':
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+        user = authenticate(username=username, password=password)
+        if user:
+            login(request,user)
+            return redirect('/')
+        else:
+            return render(request, 'registration/login.html', {'error':"Invalid UserName or Password"})
+    else:
+        return render(request, 'registration/login.html', {})
+@login_required
 def logoutView(request):
-    pass
+    logout(request)
+    return redirect('/')
 
 def registerView(request):
-    pass
\ No newline at end of file
+    if request.method=="POST":
+        try:
+            username=request.POST['username']
+            password1=request.POST['password1']
+            password2=request.POST['password2']
+            firstname=request.POST['firstname']
+            lastname=request.POST['lastname']
+            email=request.POST['email']
+        except:
```

Comment:
As a good coding practice, whenever you use try-except block, capture only the exceptions which you want to catch (IndexError, IntegrityError, etc.)